src/parser/parser.yy

```diff
@@ -701,10 +707,41 @@ drop_edge_sentence
     }
     ;
 
+order_factor
+    : input_ref_expression {
+        $$ = new OrderFactor($1, OrderFactor::ASCEND);
+    }
+    | input_ref_expression KW_ASC {
+        $$ = new OrderFactor($1, OrderFactor::ASCEND);
+    }
+    | input_ref_expression KW_DESC {
+        $$ = new OrderFactor($1, OrderFactor::DESCEND);
+    }
+    ;
+
+order_factors
+    : order_factor {
+        auto factors = new OrderFactors();
+        factors->addFactor($1);
+        $$ = factors;
+    }
+    | order_factors COMMA order_factor {
+        $1->addFactor($3);
+        $$ = $1;
+    }
+    ;
+
+order_by_sentence
+    : KW_ORDER KW_BY order_factors {
+        $$ = new OrderBySentence($3);
+    }
+    ;
```

Comment:
IMO, it should be order_by_clause, not order_by_sentence.

Reply:
i was discussing this with dutor and sheman before,  it would be better a sentence at present, or it would made the go executor too much bloated. For further reason is that we have no optimization for execution plan for now.  We should do some optimization for clause, and we'd prefer doing some go-clauses in storage layer for performance. But it is a challenge to this optimization, we'll disscuss this and put it on the agenda as soon as possible.

Reply:
I think `order by` as a separate sql is weird, it should be an algorithm  or a  clause